demo/Dockerfile

```
@@ -10,193 +10,75 @@
-# 22.04 is the current long term support release, maintained until 2025-04.
-FROM ubuntu:22.04 AS os
-
-# tzinfo hangs without this
-ARG DEBIAN_FRONTEND='noninteractive'
-
-# ---------------------------------------------------------------------
-# build
-# ---------------------------------------------------------------------
-
-FROM os AS build
-
-# limit cpus so don't run out of memory on local machine
-# symptom: get error - "c++: fatal error: Killed signal terminated program cc1plus"
-# can turn off if building in cloud
-ARG CONAN_CPU_COUNT=2
-
-ARG WITH_RUBY='True'
-
-# set some variables
-ENV PATH="$HOME/venv3.9/bin:$PATH"
-ENV CONAN_PROFILE='conan/profiles/docker'
-
-# update os and add dependencies
-# note: Dockerfiles run as root by default, so don't need sudo
-RUN  apt-get update \
-  && apt-get install -y \
-       autoconf \
-       automake \
-       build-essential \
-       cmake \
-       git \
-       python3 \
-       python3-pip \
-       rake \
-       ruby \
-  && rm -rf /var/lib/apt/lists/* \
-  && pip install conan -v 'conan==2.0.9'
-
-# make an agent directory and cd into it
-WORKDIR /root/agent
-
-# bring in the repo contents, minus .dockerignore files
-COPY . .
-
-ARG WITH_TESTS=false
-ARG WITH_TESTS_ARG=argument
-ARG SHARED=False
-
-# Build and optionally test
-RUN  if [ -z "$WITH_TESTS" ] || [ "$WITH_TESTS" = "false" ]; then \
-       WITH_TESTS_ARG="--test-folder="; \
-     else \
-       WITH_TESTS_ARG=""; \
-     fi \
-  && conan profile detect \
-  && conan create . \
-       --build=missing \
-       -c "tools.build:jobs=$CONAN_CPU_COUNT" \
-       -o agent_prefix=mtc \
-       -o cpack=True \
-       -o "with_ruby=$WITH_RUBY" \
-       -o cpack_destination=/root/agent \
-       -o cpack_name=dist \
-       -o cpack_generator=TGZ \
-       -pr "$CONAN_PROFILE" \
-       ${WITH_TESTS_ARG}
-
-# ---------------------------------------------------------------------
-# release
-# ---------------------------------------------------------------------
-
-FROM os AS release
+FROM mtconnect/agent as mtconnect
 
 # TODO: How about shortening the description to MTConnect Agent or at least MTConnect C++ Agent?
-LABEL author='mtconnect' description='MTConnect C++ Agent'
-
-ARG BIN_DIR='/usr/local/bin'
-ARG LIB_DIR='/usr/local/lib'
-
-ARG MTCONNECT_CONF_DIR='/etc/mtconnect'
-ARG MTCONNECT_DATA_DIR='/usr/local/share/mtconnect'
-ARG MTCONNECT_LOG_DIR='/var/log/mtconnect'
+LABEL author='mtconnect' description='MTConnect C++ Agent Demo'
 
-ENV MTCONNECT_CONF_DIR="$MTCONNECT_CONF_DIR"
-ENV MTCONNECT_DATA_DIR="$MTCONNECT_DATA_DIR"
-ENV MTCONNECT_LOG_DIR="$MTCONNECT_LOG_DIR"
-ENV DEMO_DIR="$MTCONNECT_DATA_DIR/demo"
-
-# install ruby for simulator
-RUN  apt-get update \
-  && apt-get install -y ruby
+USER root
 
-# change to a new non-root user for better security.
-# this also adds the user to a group with the same name.
-# -m creates a home folder, ie /home/<username>
-ARG UID=1000
-ARG GID=1000
+RUN apk add ruby
```

I think you have forgotten to update APK’s cache. You can either `-U` or `--update` or `--update-cache` with `apk add` subcommand.

```suggestion
RUN apk --update add ruby
```

---

Update was done before on the other image. I can add it in.